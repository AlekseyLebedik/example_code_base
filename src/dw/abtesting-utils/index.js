import { flatten } from 'lodash';

import { getNowTimestamp } from 'dw/core/helpers/date-time';

import { TEST_STATUS } from './constants';

export * from './constants';

export const testStartSelector = test => {
  if (test.testPeriodStart) {
    return test.testPeriodStart;
  }
  const allStartDates = flatten(
    test.cohorts.map(cohort =>
      cohort.treatments.map(treatment => treatment.start)
    )
  );
  allStartDates.push(test.catchStart);
  const startDate = Math.min(...allStartDates);
  return startDate;
};

export const testEndSelector = test => {
  if (test.testPeriodEnd) {
    return test.testPeriodEnd;
  }
  const allEndDates = test.cohorts
    .map(cohort => cohort.treatments.map(treatment => treatment.end))
    .reduce((acc, currValue) => [...acc, ...currValue], []);
  allEndDates.push(test.catchEnd);
  const endDate = Math.max(...allEndDates);
  return endDate;
};

export const determineTestStatus = test => {
  if (test.status === 'configuration') {
    return TEST_STATUS.CONFIG;
  }
  if (test.status === TEST_STATUS.LIVE) {
    const now = getNowTimestamp();
    const startDate = testStartSelector(test);
    const endDate = testEndSelector(test);
    if (now < startDate) {
      return TEST_STATUS.APPROVED;
    }
    if (startDate < now && now < endDate) {
      return TEST_STATUS.ACTIVE;
    }
    if (now > endDate) {
      return TEST_STATUS.ANALYSIS;
    }
  }
  return test.status;
};
