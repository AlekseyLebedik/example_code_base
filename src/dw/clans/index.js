import React from 'react';

import { ApolloProvider } from '@apollo/react-hooks';
import setupApollo from 'dw/core/helpers/setupApollo';

import { makeUnit } from 'dw/core/helpers/unit';
import UserReplica from 'dw/core/replicas/user';
import PermissionsReplica from 'dw/core/replicas/permissions';
import SwitchesReplica from 'dw/core/replicas/switches';
import ContentTypeReplica from 'dw/core/replicas/contentType';

import App from './components/App';

import createStore from './store';

export default makeUnit(
  <ApolloProvider client={setupApollo()}>
    <UserReplica.ConnectedReplica />
    <PermissionsReplica.ConnectedReplica />
    <SwitchesReplica.ConnectedReplica />
    <ContentTypeReplica.ConnectedReplica />
    <App />
  </ApolloProvider>,
  createStore
);
