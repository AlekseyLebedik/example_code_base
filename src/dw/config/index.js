export * from '@demonware/devzone-core/config';
