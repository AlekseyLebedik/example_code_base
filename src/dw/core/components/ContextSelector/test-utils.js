import {
  ServiceEndpoints,
  ServiceKeyMapping,
} from 'dw/online-configuration/constants';
import { CONTEXT_TYPE_TITLE, CONTEXT_TYPE_PLATFORM } from './constants';

export const DEFAULT_TITLE_CONTEXT = '5577';
export const PLATFORM_CONTEXTS = ['5577-ps4', '5577-xb1'];
export const DEFAULT_PLATFORM_CONTEXT = PLATFORM_CONTEXTS[0];

export const mockState = ({
  serviceNames = [],
  titleContext = DEFAULT_TITLE_CONTEXT,
  platformContexts = PLATFORM_CONTEXTS,
  usesMulticontext = true,
  platformEndpoints = [],
  userId,
}) => {
  const selectableContexts = Array.isArray(platformContexts)
    ? platformContexts
    : [platformContexts];
  const services = Array.isArray(serviceNames) ? serviceNames : [serviceNames];
  const usePlatform = Array.isArray(platformEndpoints)
    ? platformEndpoints
    : [platformEndpoints];
  const state = {
    user: {
      projects: [
        {
          titles: [
            {
              id: 5577,
              environments: [
                { shortType: 'dev', options: { is_multicontext: true } },
              ],
            },
          ],
        },
      ],
    },
    Components: {
      App: {
        currentTitleEnv: {
          usesMulticontext,
        },
      },
      ContextSelector: {
        Available: services.reduce(
          (acc, serviceName) => ({
            ...acc,
            [serviceName]: {
              data: [
                ...selectableContexts.map((context, idx) => ({
                  id: idx + 1,
                  name: context,
                  userSelectable: true,
                  type: CONTEXT_TYPE_PLATFORM,
                })),
                {
                  id: selectableContexts.length + 1,
                  name: titleContext,
                  userSelectable: false,
                  type: CONTEXT_TYPE_TITLE,
                },
              ],
            },
          }),
          {}
        ),
        userId,
        Registry: services.reduce(
          (acc, serviceName) => ({
            ...acc,
            [serviceName]: {
              data: Object.values(
                ServiceEndpoints[ServiceKeyMapping[serviceName]]
              ).map(endpoint => ({
                endpoint,
                type: usePlatform.includes(endpoint)
                  ? CONTEXT_TYPE_PLATFORM
                  : CONTEXT_TYPE_TITLE,
              })),
            },
          }),
          {}
        ),
      },
      TitleSelector: {
        currentTitle: { id: 5577 },
        currentEnv: { shortType: 'dev' },
      },
    },
  };
  return state;
};
