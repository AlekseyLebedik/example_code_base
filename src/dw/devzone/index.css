@tailwind utilities;

/* Enable for layout debugging */
/* * {
  outline: 1px solid red !important;
} */

*,
::before,
::after {
  box-sizing: border-box;
}

body {
  margin: 0;
  padding: 0;
  font-family: Roboto, -apple-system, BlinkMacSystemFont, 'Segoe UI',
    'Helvetica Neue', Helvetica, 'PingFang SC', 'Hiragino Sans GB',
    'Microsoft YaHei', SimSun, sans-serif;
}

body.print {
  padding-left: 0;
}

a:-webkit-any-link {
  text-decoration: none;
}

.row-flex {
  display: flex;
  flex-direction: row;
  flex: 1;
  align-self: stretch;
}

.main-container {
  overflow: hidden;
}

.main-container .empty {
  min-height: 300px;
  font-size: 1.75rem;
  text-align: center;
  margin-top: 100px;
}

.common-modal-dialog > div > div:first-child {
  width: 500px !important;
}

/* ANT TABS */

.details__container .ant-tabs,
.main-container .ant-tabs {
  height: 100%;
  overflow: hidden;
}

.details__container .ant-tabs-content,
.main-container .ant-tabs-content {
  height: calc(100% - 60px);
}

.details__container .ant-tabs-tabpane,
.main-container .ant-tabs-tabpane {
  height: 100%;
}

.scrollable-content {
  height: 100%;
  overflow-x: hidden;
  overflow-y: auto;
}

.flex-rows-container {
  display: flex;
  flex-direction: column;
  height: 100%;
  overflow: hidden;
}

/* END ANT TABS */
