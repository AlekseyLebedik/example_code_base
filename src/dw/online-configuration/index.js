import React from 'react';
import { Route, Switch } from 'react-router-dom';

import { makeUnit } from 'dw/core/helpers/unit';
import WaitFor from 'dw/core/components/WaitFor';
import UserReplica from 'dw/core/replicas/user';
import PermissionsReplica from 'dw/core/replicas/permissions';
import SwitchesReplica from 'dw/core/replicas/switches';
import ContentTypeReplica from 'dw/core/replicas/contentType';

import sourceSelectSaga from 'dw/core/components/SourceSelect/saga';
import RedirectToDefault from 'dw/core/components/RedirectToDefault';

import { projectsSelector } from 'dw/core/helpers/title-env-selectors';

import App from './components/App';

import createStore from './store';

import componentsSagas from './components/sagas';
import scenesSagas from './scenes/sagas';

const userProfileSelector = state => state.user.profile;

export default makeUnit(
  <>
    <UserReplica.ConnectedReplica />
    <PermissionsReplica.ConnectedReplica />
    <SwitchesReplica.ConnectedReplica />
    <ContentTypeReplica.ConnectedReplica />

    {/* Wait for the replicated data to be set in the store */}
    <WaitFor selectors={[userProfileSelector, projectsSelector]}>
      <Switch>
        <Route path="/online-configuration/:titleId/:env" component={App} />
        <Route component={RedirectToDefault} />
      </Switch>
    </WaitFor>
  </>,
  createStore,
  {
    sagas: [...componentsSagas, ...scenesSagas, sourceSelectSaga],
  }
);
