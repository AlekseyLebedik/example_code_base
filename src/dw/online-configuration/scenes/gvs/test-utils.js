/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow } from 'enzyme';
import { MockedProvider } from '@apollo/client/testing';
import createStore from 'dw/online-configuration/store';

import { mocks as baseMocks } from './graphql/mocks';

jest.mock('react-redux', () => {
  return {
    ...jest.requireActual('react-redux'),
    useDispatch: jest.fn(),
    useSelector: jest.fn(),
  };
});

const {
  useSelector: mockUseSelector,
  useDispatch: mockUseDispatch,
} = require('react-redux');

const setupStore = ({ scopeURI }) => {
  const { store: mockStore } = createStore();
  mockUseDispatch.mockImplementation(
    () => action => mockStore.dispatch(action)
  );
  mockStore.dispatch({
    type: 'APP_FETCH_TITLE_ENVIRONMENT_SUCCESS',
    env: {
      options: {
        scopeURI,
      },
    },
  });
  mockStore.dispatch({
    type: 'TITLE_SELECTOR.CHANGE_TITLE',
    title: { name: 'BO5 CrossPlay Scratch Dev' },
    project: { name: 'Call of Duty: Black Ops 5' },
  });
  mockUseSelector.mockImplementation(selector =>
    selector(mockStore.getState())
  );
};

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: jest.fn(),
  useLocation: jest.fn(),
  useRouteMatch: jest.fn(),
  useHistory() {
    return {
      push: jest.fn(),
    };
  },
}));

const {
  useParams: mockUseParams,
  useLocation: mockUseLocation,
  useRouteMatch: mockUseRouteMatch,
} = require('react-router-dom');

export const setupUseParams = (props, search) => {
  mockUseParams.mockReturnValue({
    ...props,
  });
  mockUseLocation.mockReturnValue({
    search,
  });
  mockUseRouteMatch.mockReturnValue({
    params: props,
  });
};

jest.mock('@apollo/client', () => ({
  ...jest.requireActual('@apollo/client'),
  useQuery: jest.fn(),
  useMutation: jest.fn(),
  useApolloClient: jest.fn(),
}));
const { useQuery, useMutation, useApolloClient } = require('@apollo/client');

jest.mock('@demonware/devzone-core/modules/user/hooks', () => ({
  useUserProfileActions() {
    return { user: { profile: { id: 1, name: 'testuser' } } };
  },
}));

jest.mock('dw/core/components/TitleEnvSelect/hooks');

const { useEnvironments } = require('dw/core/components/TitleEnvSelect/hooks');

const mockError = jest.fn();
const mockSuccess = jest.fn();
jest.mock('dw/core/hooks', () => ({
  ...jest.requireActual('dw/core/hooks'),
  useCurrentEnvPermission: jest.fn(),
  useSnackbar: jest.fn(),
}));

const { useCurrentEnvPermission, useSnackbar } = require('dw/core/hooks');

export const renderWithApolloClient = (
  Component,
  { params, scopeURI, search, extraMocks = [], permission = true, snackbar }
) => {
  useEnvironments.mockReturnValue({
    loading: false,
    environments: [
      {
        title: { id: String(params?.titleId || 1) },
        environment: { shortType: params?.env || 'dev', options: { scopeURI } },
      },
    ],
  });
  useCurrentEnvPermission.mockReturnValue(permission);
  useSnackbar.mockReturnValue({
    error: snackbar?.error || mockError,
    success: snackbar?.success || mockSuccess,
  });
  setupStore({ scopeURI });
  setupUseParams(params, search);
  const mocks = [...baseMocks, ...extraMocks];
  const wrapper = shallow(
    <MockedProvider mocks={mocks} addTypename={false}>
      {Component}
    </MockedProvider>
  );

  const { client } = wrapper.props();
  const { useQuery: useOrigQuery, useMutation: useOrigMutation } =
    jest.requireActual('@apollo/client');
  useQuery.mockImplementation((q, options) => {
    const result = useOrigQuery(q, {
      ...options,
      client,
      fetchPolicy: 'cache-first',
    });
    return { ...result, refetch: jest.fn() };
  });
  useMutation.mockImplementation((m, options) =>
    useOrigMutation(m, { ...options, client })
  );
  useApolloClient.mockReturnValue(client);
  return wrapper.shallow().shallow().shallow().shallow();
};
