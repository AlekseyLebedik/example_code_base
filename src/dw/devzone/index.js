// Disabled because we want to import theming and global CSS first
/* eslint-disable import/first */

import 'normalize.css';
import '@demonware/devzone-core/themes/default.css';
import defaultTheme from '@demonware/devzone-core/themes';

import './index.css';

import React from 'react';
import ReactDOM from 'react-dom';
import { Router } from 'react-router-dom';
import 'regenerator-runtime/runtime';

import { LicenseManager } from 'ag-grid-enterprise';

// eslint-disable-next-line
import * as monacoEditor from 'monaco-editor';
import MonacoEditor from 'react-monaco-editor';

import {
  createGenerateClassName,
  StylesProvider,
  MuiThemeProvider,
} from '@material-ui/core/styles';
import ReactGA from 'react-ga';

import history from 'dw/core/helpers/history';
import { redirectIfUnauthenticated } from '@demonware/devzone-core/auth';
import { AppWrapper } from '@demonware/devzone-core';
import { rootSaga } from '@demonware/devzone-core/helpers/sagas';
import { GA_PROPERTY_ID, AG_LICENSE_KEY } from 'dw/config';

import Devzone from './unit';

import 'dw/core/helpers/polyfills';

LicenseManager.setLicenseKey(AG_LICENSE_KEY);

window.monacoEditor = monacoEditor;
window.MonacoEditor = MonacoEditor;

function setupGA() {
  ReactGA.initialize(GA_PROPERTY_ID, { siteSpeedSampleRate: 100 });

  // Track initial page load
  ReactGA.pageview(window.location.pathname + window.location.search);

  // Track every page change
  history.listen(location =>
    ReactGA.pageview(location.pathname + location.search)
  );
}

const generateClassName = createGenerateClassName({ productionPrefix: 'dz' });

// Is the user authenticated? Or should we redirect to the login page?
redirectIfUnauthenticated(history)
  .then(({ isSilentAuth }) => {
    if (isSilentAuth) {
      /* eslint-disable-next-line */
      console.log('Silent Authentication callback.');
      return;
    }

    setupGA();

    ReactDOM.render(
      <StylesProvider generateClassName={generateClassName}>
        <MuiThemeProvider theme={defaultTheme}>
          <AppWrapper rootSaga={rootSaga}>
            <Router history={history}>
              <Devzone />
            </Router>
          </AppWrapper>
        </MuiThemeProvider>
      </StylesProvider>,
      document.getElementById('root')
    );
  })
  .catch(() => {
    // eslint-disable-next-line
    console.log('Redirecting to Authentication Provider...');
  });
