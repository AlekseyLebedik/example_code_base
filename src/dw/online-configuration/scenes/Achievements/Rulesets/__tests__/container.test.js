import React from 'react';
import createRouterContext from 'react-router-test-context';
import { BrowserRouter as Router } from 'react-router-dom';

import { shallowUntilTarget } from 'dw/test-utils';
import createStore from 'dw/online-configuration/store';

import ConnectedRulesets from '../container';

describe('PlayerAssets', () => {
  let store;
  let context;

  const renderConnected = (customProps = {}) => {
    ({ store } = createStore());
    context = createRouterContext({
      match: {
        url: '/achievements/rulesets',
      },
      path: '/achievements/rulesets/:id?',
    });
    const onLoad = jest.fn();
    const onShowMore = jest.fn();
    const propsConnected = {
      store,
      onLoad,
      onShowMore,
      ...customProps,
    };

    return shallowUntilTarget(
      <Router>
        <ConnectedRulesets {...propsConnected} />
      </Router>,
      'Connect(Rulesets)',
      {
        shallowOptions: { context },
      }
    );
  };

  describe('Connected Ruleset Assets', () => {
    it('renders correctly', () => {
      const wrapper = renderConnected();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
