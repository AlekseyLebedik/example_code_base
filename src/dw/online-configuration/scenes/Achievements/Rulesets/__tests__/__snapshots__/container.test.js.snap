// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerAssets Connected Ruleset Assets renders correctly 1`] = `
<ContextProvider
  value={null}
>
  <Rulesets
    deleteRulesetEvents={
      Object {
        "onDeleteRulesetHandler": [Function],
      }
    }
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    match={
      Object {
        "isExact": true,
        "params": Object {},
        "path": "/",
        "url": "/",
      }
    }
    onClickListItem={[Function]}
    onLoad={[Function]}
    onResetSelectedRuleset={[Function]}
    onSearch={[Function]}
    onShowMore={[Function]}
    rulesets={Array []}
    uploadRulesetEvents={
      Object {
        "closeUploadRulesetModalHandler": [Function],
        "onUploadRulesetHandler": [Function],
        "openUploadRulesetModalHandler": [Function],
        "uploadOnRemoteSubmit": [Function],
      }
    }
  />
</ContextProvider>
`;
